bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/handler/SoulissGatewayHandler.java

```diff
@@ -0,0 +1,324 @@
+import org.openhab.binding.souliss.internal.SoulissDatagramSocketFactory;
+import org.openhab.binding.souliss.internal.protocol.SoulissBindingNetworkParameters;
+import org.openhab.binding.souliss.internal.protocol.SoulissBindingSendDispatcherJob;
+import org.openhab.binding.souliss.internal.protocol.SoulissBindingUDPServerJob;
+import org.openhab.binding.souliss.internal.protocol.SoulissCommonCommands;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SoulissGatewayHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Tonino Fazio - Initial contribution
+ * @author Luca Calcaterra - Refactor for OH3
+ */
+@NonNullByDefault
+public class SoulissGatewayHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SoulissGatewayHandler.class);
+    public @Nullable DatagramSocket datagramSocketDefaultPort;
+    private @Nullable SoulissBindingUDPServerJob udpServerDefaultPortRunnableClass;
+    private SoulissCommonCommands soulissCommands = new SoulissCommonCommands();
+
+    boolean bGatewayDetected = false;
+
+    private @NonNullByDefault({}) Configuration gwConfigurationMap;
+
+    public int pingRefreshInterval;
+    public int subscriptionRefreshInterval;
+    public boolean thereIsAThingDetection = true;
+    public int healthRefreshInterval;
+    public int sendRefreshInterval;
+    public int sendTimeoutToRequeue;
+    public int sendTimeoutToRemovePacket;
+    private Bridge bridge;
+    public int preferredLocalPort;
+    public int soulissGatewayPort;
+    public byte userIndex;
+    public byte nodeIndex;
+    public String ipAddressOnLAN = "";
+    private int nodes;
+    private int maxTypicalXnode;
+    private int countPingKo = 0;
+
+    public SoulissGatewayHandler(Bridge br) {
+        super(br);
+        bridge = br;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initializing server handler for thingGeneric {}", getThing());
+
+        gwConfigurationMap = bridge.getConfiguration();
+        ipAddressOnLAN = (String) gwConfigurationMap.get(SoulissBindingConstants.CONFIG_IP_ADDRESS);
+        if (gwConfigurationMap == null) {
+            logger.debug("Gateway Handler - Error in Configuration Map - Thing= {}", getThing());
+        } else {
+            if (gwConfigurationMap.get(SoulissBindingConstants.CONFIG_LOCAL_PORT) != null) {
```

Comment:
The config processing could be reduced to a couple of lines. You could utilize `getConfigAs()`. The parameters are validated by the framework.

Reply:
reworked gw Config (as wrote above)